### PR TITLE
refactor: misc cleanup

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: src/angularfire2.ts [hunk 1]
```diff
@@ -26,53 +26,35 @@ export class AngularFire {
...
+export function authBackendFactory(ref: Firebase): FirebaseSdkAuthBackend {
   return new FirebaseSdkAuthBackend(ref, false);
 }
 
 /**
- * Used to define the default Firebase root location to be
- * used throughout an application.
+ * Used to define the default Firebase root location to be used throughout an application.
+ *
+ * @deprecated use `{provide: FirebaseUrl, useValue: 'url'}` to enable angular offline compilation
```
Comment (vicb) [inline src/angularfire2.ts]: !

File: src/providers/auth.ts [hunk 2]
```diff
@@ -19,10 +16,11 @@ import {
 
 const kBufferSize = 1;
 
-export const firebaseAuthConfig = (config: AuthConfiguration): Provider => {
-  return provide(FirebaseAuthConfig, {
-    useValue: config
-  });
+/**
+ * @deprecated use `{provide: FirebaseAuthConfig, useValue: config}` to enable angular offline compilation
```
Comment (vicb) [inline src/providers/auth.ts]: !

**Comment by jeffbcross** [comment]
I'll wait to merge until after #200 lands.

**Comment by vicb** [comment]
Do you have a timeline (this PR is needed to have offline compilation) ?

**Comment by davideast** [comment]
Closing due to the drastic changes in the project.

**Pull request closed without merging**